cmake/external_dependencies.cmake

```diff
@@ -83,20 +83,29 @@ if(NOT NO_HTTP_CLIENT AND NOT USE_CRT_HTTP_CLIENT)
             include(FindCURL)
             if(NOT CURL_FOUND)
                 message(FATAL_ERROR "Could not find curl")
-            else()
+            endif()
+
+            # When built from source using cmake, curl does not include
+            # CURL_INCLUDE_DIRS or CURL_INCLUDE_DIRS so we need to use
+            # find_package to fix it
+            if ("${CURL_INCLUDE_DIRS}" STREQUAL "" AND "${CURL_LIBRARIES}" STREQUAL "")
+                message(STATUS "Could not find curl include or library path, falling back to find with config.")
+                find_package(CURL)
+                set(CURL_LIBRARIES CURL::libcurl)
+                set(CLIENT_LIBS_ABSTRACT_NAME curl)
```

Comment:
nit: As this happens in both branches of the `if` it might be worth pulling it out as it always happens